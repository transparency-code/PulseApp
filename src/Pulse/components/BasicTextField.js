//https://material-ui.com/components/text-fields/
//https://material-ui.com/components/text-fields/#layout
import React from 'react';
import TextField from '@material-ui/core/TextField';


//controlled component. has value and handle Change
export default function BasicTextField({label,type="text",valueProp,handleChangeProp}) {

//  console.log(fullWidth)

  const handleChange = event => {
    let changedValue = event.target.value
    changedValue = type === "number" ? Number(changedValue) : changedValue
    handleChangeProp(changedValue)
  };

  return (

       <TextField id={label} variant="standard" label={label} type={type}  fullWidth value={valueProp} onChange={handleChange} />

  );
}