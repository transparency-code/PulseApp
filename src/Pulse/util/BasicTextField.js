//https://material-ui.com/components/text-fields/
//https://material-ui.com/components/text-fields/#layout
import React from 'react';
import TextField from '@material-ui/core/TextField';


//controlled component. has value and handle Change
export default function BasicTextField({label,type="text",valueProp,handleChangeProp}) {

//  console.log(fullWidth)

  const handleChange = event => {
    const changedValue = event.target.value
    handleChangeProp(changedValue)
  };

  return (

       <TextField id={label} variant="standard" label={label} type={type}  fullWidth value={valueProp} onChange={handleChange} />

  );
}