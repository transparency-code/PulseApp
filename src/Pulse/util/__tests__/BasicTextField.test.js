//https://material-ui.com/components/text-fields/

import React from "react";
import BasicTextField from "../BasicTextField";
import renderer from "react-test-renderer";
import { render, fireEvent, cleanup } from "@testing-library/react";

afterEach(cleanup);

it("snapshot testing  - default type text", () => {

    const aTextField = {
        label: "aLabel",
      }

  const mockCallback = jest.fn();

  const tree = renderer
    // {label,checkedProp,onCheckedChange}
    .create(<BasicTextField {...aTextField} valueProp={"hello"} handleChangeProp={mockCallback} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("snapshot testing  - type number", () => {
  const aTextField = {
    label: "aLabel",
    type: "number"
  };

  const mockCallback = jest.fn();

  const tree = renderer
    // {label,checkedProp,onCheckedChange}
    .create(
      <BasicTextField
        {...aTextField}
        valueProp={"hello"}
        handleChangeProp={mockCallback}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("fire a change - type text - text change success", () => {
  const aTextField = {
    label: "aLabel",
  };

  const mockCallback = jest.fn();

  const { getByLabelText } = render(
    <BasicTextField
      {...aTextField}
      checkedProp={true}
      handleChangeProp={mockCallback}
    />
  );

  const selectNode = getByLabelText("aLabel");

  ////do a fireEvent.change
  fireEvent.change(selectNode, { target: { value: "newValue" } });

  expect(mockCallback.mock.calls).toHaveLength(1);
  expect(mockCallback).toHaveBeenCalledWith("newValue");
});

it("fire a change - type number - number change success", () => {
    const aTextField = {
      label: "aLabel",
    };
  
    const mockCallback = jest.fn();
  
    const { getByLabelText } = render(
      <BasicTextField
        {...aTextField}
        checkedProp={true}
        handleChangeProp={mockCallback}
      />
    );
  
    const selectNode = getByLabelText("aLabel");
  
    ////do a fireEvent.change
    fireEvent.change(selectNode, { target: { value: "20" } });
  
    expect(mockCallback.mock.calls).toHaveLength(1);
    expect(mockCallback).toHaveBeenCalledWith("20");
  });
  
  it("fire a change - type number - text change fails", () => {
    const aTextField = {
      label: "aLabel",
      type : "number"
    };
  
    const mockCallback = jest.fn();
  
    const { getByLabelText } = render(
      <BasicTextField
        {...aTextField}
        checkedProp={true}
        handleChangeProp={mockCallback}
      />
    );
  
    const selectNode = getByLabelText("aLabel");
  
    ////do a fireEvent.change
    fireEvent.change(selectNode, { target: { value: "new" } });
  
    expect(mockCallback.mock.calls).toHaveLength(0);
    
  });
  
