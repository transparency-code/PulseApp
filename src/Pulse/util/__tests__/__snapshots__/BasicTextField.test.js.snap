// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing  - default type text 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="aLabel"
    id="aLabel-label"
  >
    aLabel
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      id="aLabel"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="hello"
    />
  </div>
</div>
`;

exports[`snapshot testing  - type number 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
    htmlFor="aLabel"
    id="aLabel-label"
  >
    aLabel
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      id="aLabel"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="number"
      value="hello"
    />
  </div>
</div>
`;
